lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3OutputStream.java

```diff
@@ -272,6 +290,7 @@ private CompletedPart uploadPage(byte[] data, int length)
         if (uploadId.isEmpty()) {
             CreateMultipartUploadRequest request = CreateMultipartUploadRequest.builder()
                     .overrideConfiguration(context::applyCredentialProviderOverride)
+                    .acl(cannedAcl)
```

Comment:
set `acl` only if not `null`

Reply:
I think it's fine to always set it, as the API is designed to accept null as a way to not set an ACL on the upload.